### PR TITLE
Refactor bucketingprocess

### PR DESCRIPTION
**Pull request by timvink**

### CODE REVIEW

File: skorecard/reporting/plotting.py [hunk 1]
```diff
@@ -88,13 +90,14 @@ def plot_prebucket_table(prebucket_table, X, y, column, format=None):
     fig.update_traces(marker=dict(color=bar_colors), selector=dict(type="bar"))
 
     if format is not None:
-        img_bytes = fig.to_image(format=format)
+        img_bytes = fig.to_image(format=format, scale=2, width=1050, height=525)
```
Comment (orchardbirds) [inline skorecard/reporting/plotting.py:93]: small detail, but do we want the user to be able to edit the height and width?

File: skorecard/reporting/plotting.py [hunk 2]
```diff
@@ -145,7 +148,7 @@ def plot_bucket_table(bucket_table, format=None):
     )
 
     if format is not None:
-        img_bytes = fig.to_image(format=format)
+        img_bytes = fig.to_image(format=format, scale=2, width=1050, height=525)
```
Comment (orchardbirds) [inline skorecard/reporting/plotting.py:151]: same here

File: skorecard/pipeline/bucketing_process.py [hunk 3]
```diff
@@ -87,7 +90,7 @@ def bucket_table(self, column):
         the BucketingProcess ends up with the final buckets.
         An example is seen below:
 
-        pre-bucket | label              | Count | Count (%) | Non-event | Event | Event Rate | WoE |  IV   | bucket
+        bucket     | label              | Count | Count (%) | Non-event | Event | Event Rate | WoE |  IV   | bucket
```
Comment (orchardbirds) [inline skorecard/pipeline/bucketing_process.py:93]: shouldn't one of these be labeled 'pre-bucket'? Right now you have 'bucket' twice

File: skorecard/pipeline/bucketing_process.py [hunk 4]
```diff
@@ -230,58 +221,66 @@ def fit(self, X, y=None):
...
-        # to find the mappings of the specials for the bucketing step.
+        # Fit the prebucketing pipeline
+        # And save the prebucket mapping
+        X_prebucketed_ = self.prebucketing_pipeline.fit_transform(X, y)
+        assert isinstance(X_prebucketed_, pd.DataFrame)
         self._features_prebucket_mapping = get_features_bucket_mapping(self.prebucketing_pipeline)
-
-        # define
+        # and calculate the prebucket tables.
+        self.prebucket_tables_ = dict()
```
Comment (orchardbirds) [inline skorecard/pipeline/bucketing_process.py:230]: Really nice idea

**Review by orchardbirds — CHANGES_REQUESTED** [review]
Small comments, overall really nice work man :)

**Comment by orchardbirds** [comment]
oh and the tests are failing :(

File: skorecard/reporting/plotting.py (hunk 1 above)
Comment (timvink) [reply skorecard/reporting/plotting.py:93]: fixed

File: skorecard/pipeline/bucketing_process.py (hunk 3 above)
Comment (timvink) [reply skorecard/pipeline/bucketing_process.py:93]: yes

**Comment by timvink** [comment]
@orchardbirds fixed the tests, expanded the scope a bit of the PR to also include updates to the dash app (that uses the refactored BucketingProcess)

**Review by orchardbirds — APPROVED** [review]
I like the name 'interactive bucketer, maybe we rename the bucketer to that later? '

**Pull request merged**